### ISSUE TITLE
Command `translation:update` doesn't take into account domain

### ISSUE THREAD

**Issue by lukepass** [symfony/symfony#42285]
Hello, I am using Symfony 4.4 and I just realized that when using the following command:

`php bin/console translation:update --force en`

The command doesn't take into account the specified domain, for example:

```php
class ChangePasswordFormType extends AbstractType
{
    /**
     * @var TranslatorInterface
     */
    private $translator;

    public function __construct(TranslatorInterface $translator)
    {
        $this->translator = $translator;
    }

    /**
     * @param mixed[]                                           $options
     * @param FormBuilderInterface<FormBuilderInterface|string> $builder
     */
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $message = $this->translator->trans('This is the string I would like to translate', $parameters = [], $domain = 'validators');

        // other code...
    }
}
```

The string `This is the string I would like to translate` should be inserted in the file `validators.en.xlf` but it's written in the `messages.en.xlf` file.

Thanks!

**Comment by JohJohan** [comment]
I see you can pass `domain` as a argument to the command idk how that would effect the output?

**Comment by lukepass** [comment]
Hi @JohJohan that argument should just filter the translations of the specified domain.

**Comment by xabbuh** [comment]
Generally, the PHP extractor should be able to extract domains too. Can you create a small example application that allows to reproduce the behaviour that you experience?

**Comment by lukepass** [comment]
Of course! This is a really simple project, if you type `➜  php bin/console translation:update --force en` the result is the following:

![image](https://user-images.githubusercontent.com/6429961/127665260-20b94304-1639-4ba9-bc8c-89cedc829071.png)

(it should be inside validators)

Thanks.

Project: [translation_bug.zip](https://github.com/symfony/symfony/files/6908017/translation_bug.zip)

**Comment by xabbuh** [comment]
Thank you that helped a lot! I think the extractor is not able to parse the domain properly because of the use of the `$domain` variable.

If you change the line to something like this everything is working as expected for me:

```php
        $message = $this->translator->trans('This is the string I would like to translate', [], 'validators');
```

**Comment by lukepass** [comment]
Yes, I can confirm that if I remove `$domain = 'validators'` the string goes to validators correctly. Maybe it's related to PHP Parser and it's considered a non-constant expression?

**Comment by carsonbot** [comment]
Hey, thanks for your report!
There has not been a lot of activity here for a while. Is this bug still relevant? Have you managed to find a workaround?

**Comment by Koc** [comment]
I can confirm that issue is still present for php extractor in 5.4.x.

**Comment by nicolas-grekas** [comment]
/cc @welcoMattic 
Anyone up for having a look and fixing this?

**Comment by welcoMattic** [comment]
After discussion with @nicolas-grekas and @stof IRL, we concluded that is much more complicated to fix this issue (and not break concatenation of variable and string). So we decided to won't fix for now.

But, we noticed that we do not support named arguments in PhpExtractor. To do it, we need to write a new Extractor based on AST. Once this new Extractor will ne written, we may plan to support inline variable assignation.
